website/integrations: add Immich

ZuluWhiskey opened the pull request:
<!--
👋 Hi there! Welcome.

Please check the Contributing guidelines: https://goauthentik.io/developer-docs/#how-can-i-contribute
-->

Create documentation for the Immich application

<!--
Explain what this PR changes, what the rationale behind the change is, if any new requirements are introduced or any breaking changes caused by this PR.

Ideally also link an Issue for context that this PR will close using `closes #`
-->
Adding in documentation for Immich. This doesn't close a PR but will help anyone else attempting to set up the application in authentik

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)
-   [ ] The translation files have been updated (`make i18n-extract`)

If applicable

-   [x] The documentation has been updated
-   [x] The documentation has been formatted (`make website`)

ZuluWhiskey committed:
Create Immich integration documentation

ZuluWhiskey committed:
Add in 'title:'

rissson commented on website/integrations/services/immich/index.md:
```diff
@@ -0,0 +1,50 @@
+The following placeholders will be used:
+
+-   `immich.company` is the FQDN of the Immich install.
+-   `192.168.0.2:2283` is the local IP address & port of the Immich install.
```
```suggestion
```

Let's remove any custom environment examples, and only stick to using `immich.company`.

ZuluWhiskey replied:
I wanted to include the local IP in some way as this needs to be added to the redirect URIs to allow for local access using the IP address rather than the FQDN. Is there a way we can compromise here? Perhaps a placeholder just in the URI redirect section to say "add your local IP here too"?

ZuluWhiskey replied:
That's fair enough - Double checking the Immich documentation, it does just show the local IP as an example along with `localhost` too
```
Redirect URIs should contain all the domains you will be using to access Immich. Some examples include
```
So I'll try incorporate this wording into the URI section here too.

rissson commented on website/integrations/services/immich/index.md:
```diff
@@ -0,0 +1,50 @@
+	- **Client Secret**: Either create your own Client Secret or make a note of the auto-populated one
+	- **Redirect URIs/Origins (RegEx)**:
+	  - app.immich:/
+	  - http://192.168.0.2:2283/auth/login
+	  - http://192.168.0.2:2283/user-settings
```
```suggestion
```

Same as above.

rissson commented on website/sidebarsIntegrations.js:
```diff
@@ -105,6 +105,7 @@ module.exports = {
                         "services/sonarr/index",
                         "services/tautulli/index",
                         "services/weblate/index",
+                        "services/immich/index",
```
We try to keep this list alphabetically sorted.

ZuluWhiskey replied:
Sorry, makes sense!

rissson reviewed (changes requested):
Hi, thank you for your contribution! A few minor changes, but all in all looks very good :D (and now I don't have to figure out Immich settings when I'll install it at home)

rissson commented:
Looks like you're missing a make website. If you do not have the tooling, I can run it for you and push it to this PR.

ZuluWhiskey committed:
Move Immich so the list is alphabetical

ZuluWhiskey committed:
- Remove reference to local IPs for Redirect URIs/Origins (RegEx). Explain the list is an example
- Fixed a typo with the word "Confidential"
- Changed placeholder for Immich to be the URL, including https://

ZuluWhiskey commented:
> Looks like you're missing a make website. If you do not have the tooling, I can run it for you and push it to this PR.

Apologies, this is new to me and I'm getting this error and I'm unsure how to resolve it ☹️ If you are able to run it, I'd appreciate it
```
zuluwhiskey@xxx:~/github/authentik/authentik/website$ nodejs make website-install
node:internal/modules/cjs/loader:1147
  throw err;
  ^

Error: Cannot find module '/home/zuluwhiskey/github/authentik/authentik/website/make'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
    at Module._load (node:internal/modules/cjs/loader:985:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v20.10.0
```

rissson commented:
No worries. The command is simply `make website`, without `nodejs` in front.

ZuluWhiskey commented:
> No worries. The command is simply `make website`, without `nodejs` in front.

I should've said that I tried that but got this...
```
zuluwhiskey@x:~/github/authentik/authentik/website$ make website
make: *** No rule to make target 'website'.  Stop.
zuluwhiskey@x:~/github/authentik/authentik/website$ make website-install
make: *** No rule to make target 'website-install'.  Stop.
```
So tried it with nodejs in front as the documentation says it's required

ZuluWhiskey committed:
Changes made following "make website"

ZuluWhiskey commented:
Worked it out, was trying to run the `make website` command from the `/website` directory rather than the top directory. Changes pushed

tanberry commented on website/integrations/services/immich/index.md:
```diff
@@ -0,0 +1,48 @@
+-   `https://immich.company` is the URL used to access the Immich instance.
+-   `authentik.company` is the FQDN of the authentik install.
+
+## authentik Configuration
```
```suggestion
## authentik configuration
```

tanberry replied:
Sorry I know it was capitalized in the template; I have fixed that now. :-)

ZuluWhiskey replied:
Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

tanberry commented on website/integrations/services/immich/index.md:
```diff
@@ -0,0 +1,48 @@
+    - **Client ID**: Either create your own Client ID or make a note of the auto-populated one
+    - **Client Secret**: Either create your own Client Secret or make a note of the auto-populated one
+    - **Redirect URIs/Origins (RegEx)**:
+      _Please note that the following URIs are just examples. Ensure to include all of the domains / URLs you will be using to access Immich._
```
```suggestion
      _Please note that the following URIs are just examples. Be sure to include all of the domains / URLs that you will use to access Immich._
```

ZuluWhiskey replied:
Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

tanberry commented on website/integrations/services/immich/index.md:
```diff
@@ -0,0 +1,48 @@
+        - https://immich.company/user-settings
+    - **Signing Key**: authentik Self-signed Certificate
+    - Leave everything else as default
+2. Open the new provider you've just created
```
```suggestion
2. Open the new provider you've just created.
```

ZuluWhiskey replied:
Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

tanberry commented on website/integrations/services/immich/index.md:
```diff
@@ -0,0 +1,48 @@
+    - **Signing Key**: authentik Self-signed Certificate
+    - Leave everything else as default
+2. Open the new provider you've just created
+3. Make a note of the **OpenID Configuration Issuer**
```
```suggestion
3. Make a note of the **OpenID Configuration Issuer**.
```

ZuluWhiskey replied:
Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

tanberry commented on website/integrations/services/immich/index.md:
```diff
@@ -0,0 +1,48 @@
+
+## authentik Configuration
+
+1. Create a new OAuth2/OpenID Provider
```
```suggestion
1. Create a new OAuth2/OpenID Provider using the following settings:
```

ZuluWhiskey replied:
Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

tanberry commented on website/integrations/services/immich/index.md:
```diff
@@ -0,0 +1,48 @@
+2. Open the new provider you've just created
+3. Make a note of the **OpenID Configuration Issuer**
+
+## Service Configuration
```
```suggestion
## Immich configuration
```

tanberry commented on website/integrations/services/immich/index.md:
```diff
@@ -0,0 +1,48 @@
+
+Immich documentation can be found here: https://immich.app/docs/administration/oauth
+
+1. Go to Administration > Settings > OAuth Authentication
```
```suggestion
1. In Immich, navigate to **Administration** > **Settings** > **OAuth Authentication**.
```

ZuluWhiskey replied:
Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

tanberry commented on website/integrations/services/immich/index.md:
```diff
@@ -0,0 +1,48 @@
+1. Go to Administration > Settings > OAuth Authentication
+2. Configure Immich as follows:
+    - **Issuer URL**: Populate this field with the `OpenID Configuration Issuer`
+    - **Client ID**: Enter yout Client ID from authentik here
```
```suggestion
    - **Client ID**: Enter your Client ID from authentik
```

ZuluWhiskey replied:
Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

tanberry commented on website/integrations/services/immich/index.md:
```diff
@@ -0,0 +1,48 @@
+2. Configure Immich as follows:
+    - **Issuer URL**: Populate this field with the `OpenID Configuration Issuer`
+    - **Client ID**: Enter yout Client ID from authentik here
+    - **Client Secret**: Enter yout Client Secret from authentik here
```
```suggestion
    - **Client Secret**: Enter your Client Secret from authentik
```

ZuluWhiskey replied:
Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

tanberry reviewed (approved):
Thanks so much for this contribution! Just a few grammar nitpicks, punctuation, but LGTM!

ZuluWhiskey committed:
Fixing grammar and typos

ZuluWhiskey commented on website/integrations/services/immich/index.md:
```diff
@@ -0,0 +1,48 @@
+
+## What is Immich
+
+> Immich is a self-hosted backup solution for photos and videos on mobile devices
```
Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

ZuluWhiskey commented:
> Thanks so much for this contribution! Just a few grammar nitpicks, punctuation, but LGTM!

Thank you, I'll keep the wording / language in mind for an future PRs 😄 All issues mentioned fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)
